polymode-compat.el

```diff
@@ -371,6 +371,86 @@ changes."
 (pm-around-advice #'font-lock-extend-region-multiline
                   #'pm-check-for-real-change-in-extend-multiline)
 
+
+;;; General protective localization and relaxed mode installation
+
+(defvar-local pm--localizing-span nil
+  "Global localizing advices related to `polymode'
+will be active whenever the value is bound to a non-nil value
+which is then presumed to be a polymode span object.
+
+For internal use only, so far.")
+
+(defun pm--localizing-narrow (f &rest args)
+  (if pm--localizing-span
+      (save-restriction
+        (narrow-to-region (nth 1 pm--localizing-span)
+                          (nth 2 pm--localizing-span))
+        (apply f args))
+    (apply f args)))
```

Comment:
If narrowing to current span is all what is needed, then  one could just do:

```
(pm-around-advice 'some-function #'pm-execute-narrowed-to-span)
```

Reply:
Thank you.  Yes, almost certainly, significant rewrites are needed.  I submitted what seemed to be working for me just because it's a better start than an issue without any patch.  Hopefully, I'll rewrite it soon.